### ISSUE TITLE
Busy dialog + kodi crashes

### ISSUE THREAD

**Issue by henryjfry** [xbmc/xbmc#20297]
### Describe the bug

This particular issue can be reproduced if you are using the plugin.program.autocompletion v2.0.2 from my repo (or the official repo) and appears to be a timing issue with the busy dialog and a jsonprc call for "Input.SendText"
If you dont close the busy dialog before you try and send the text to the input box kodi will completely crash, with no errors to the log, it just stops:

On android it requires a 250ms pause between closing the busy dialog and then the Input.SendText event, without this pause kodi will consistently crash every time you select an autocomplete option.

## Expected Behavior
The busy dialog should close on its own when an event occurs which would otherwise be blocked by the busy dialog.

Kodi should not completely crash without any particular messages to the log to describe what the actuall error which occurred was.

## Actual Behavior
Selecting the autocomplete option causes the busy dialog to load, if you try and send the inputsendtext event while the busy dialog is on screen kodi crashes.
If you close the busy dialog first, but dont have a sufficiently long pause before the input send text event, kodi crashes.

The busy dialog also casues issues for me with youtube, videos load the busy dialog right before they start playing. It will close when the video starts up, but just having the busy dialog open prior to "creating demuxer" seems to cause the video load in a small section of the screen, it will then go full screen but the video will then be very jerky and buffer contstantly.
Tailing the log and forcing the busy dialog to close and fullscreenvideo to activate immediately as soon as "VideoPlayer::OpenFile:" appears in the logs seems to stop that particular issue, but again if the busy dialog is a blocker to fullscreenvideo opening, it should close automatically first and then the video should start playing.

## Possible Fix
Make the busy dialog event less blocking?
Have kodi more reliably close the busy dialog and verify its fully closed before it starts the next action?

Maybe have the busy dialog close and reopen when other events happen in kodi?
But the busy dialog can and will cause kodi to crash in strange ways so perhaps it needs completely rebuilt in a new way so that it closes more reliably.

### To Reproduce
Using plugin.program.autocompletion on android v2.0.1/2.0.2 from the kodi repo, start typing, select an autocomplete option, kodi crashes.
If there are xbmc.sleep pauses in "~/.kodi/addons/plugin.program.autocompletion/plugin.py" after the "xbmc.executebuiltin('Dialog.Close(busydialognocancel)')", remove the pauses and it will consistently crash every time.

Seems to be similar to this problem:
https://github.com/xbmc/xbmc/issues/19304
(Note the youtube mention in that issue)

### Debuglog
(logs dont help because it crashes before anything is logged).
https://github.com/henryjfry/repository.thenewdiamond/files/7319133/kodi.log

### Your Environment

 - [x ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
Kodi 19.2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by malard** [comment]
*This reply was written and posted by Claude on @malard's behalf.*

Worth a retest before this goes any further. The mechanism you describe - a busy dialog open when another operation arrives, needing a 250 ms gap to avoid a crash - is the shape of the re-entrancy defect fixed by xbmc/xbmc#27038, merged 2025-08-11, four years after you filed this.

`CGUIDialogBusy::WaitOnEvent` now counts waiters rather than assuming a single one (`xbmc/dialogs/GUIDialogBusy.cpp:83-87`):

```cpp
const uint32_t level = ++dialog->m_waiters;
if (level == 1)
  dialog->Open();
```

Previously a second entrant re-opened the dialog and stacked a second nested render loop on top of the first, which is consistent with an abrupt exit and nothing in the log.

That is not a claim that it fixes this - your repro needs a specific add-on and a specific timing window, and nobody has run it. But it is the most likely candidate by some distance, and this report predates the fix by four years.

@henryjfry @ITguy3118 - if either of you can still reproduce this, a v22 build is the test that matters. If the 250 ms pause is no longer needed, this can be closed.

Labelling `Component: GUI engine` and `Component: JSON-RPC`.
